src/Dropdown.php

```diff
@@ -3896,6 +3896,10 @@ public static function getDropdownUsers($post, $json = true)
             $post['page_limit'] = $CFG_GLPI['dropdown_max'];
         }
 
+        if (isset($post['_one_id'])) {
+            $post['page_limit'] = -1;
```

Comment:
Mabe the problem here is that `_one_id` is only handled on server side for `CommonTreeDropdown`. By putting the `page_limit` to -1, are your loading all users ?

Reply:
Yes, `page_limit = -1` load all users

Reply:
So, maybe the solution is to return only the matching user when `_one_id`, instead of fetching users using `User::getSqlSearchResult()`